Common/Orders/Fills/LatestPriceFillModel.cs

```diff
@@ -71,20 +71,23 @@ protected override Prices GetPrices(Security asset, OrderDirection direction)
             // Assume that the security has both a trade and quote subscription
             // This should be true for crypto securities
             var quoteBar = asset.Cache.GetData<QuoteBar>();
-            var tradeBar = asset.Cache.GetData<TradeBar>();
-
-            if (tradeBar.EndTime > quoteBar.EndTime)
-            {
-                // The latest pricing data came from a trade
-                return new Prices(tradeBar);
-            }
-            else
+            if (quoteBar != null)
             {
-                // The latest pricing data came from a quote
-                var bar = direction == OrderDirection.Sell ? quoteBar.Bid : quoteBar.Ask;
-                if (bar != null)
+                var tradeBar = asset.Cache.GetData<TradeBar>();
+
+                if (tradeBar.EndTime > quoteBar.EndTime)
```

Comment:
should we also be checking if `tradeBar` is null?

Reply:
I thought it would be covered but proved wrong by unit tests that I have just added. Thank you!